### PR TITLE
Create UI

### PR DESCRIPTION
**Pull request by MaxHolmes142857**
## Related Tickets
- [#Task 79519: Content](https://edu-redmine.sun-asterisk.vn/issues/79519)
## WHAT
- 
## Evidence (Screenshot or Video)
Screenshot_20240808_103809

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

**Review by xuanthu2002 — APPROVED** [review]

**Review by TranTinh030892 — APPROVED** [review]

File: app/src/main/res/layout/fragment_home.xml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        xmlns:tools="http://schemas.android.com/tools"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".screen.MainActivity">
+
+    <TextView
+            android:id="@+id/tvHeader"
+            android:layout_width="@dimen/dp_0"
+            android:layout_height="wrap_content"
+            android:background="#E4DFDF"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: Đặt vào file color em ơi

File: app/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            android:padding="@dimen/dp_16"
+            android:text="@string/app_name"
+            android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                    android:layout_height="match_parent"
+                    android:layout_weight="1"
+                    android:orientation="horizontal"
+                    >
+                <TextView
+                        android:id="@+id/tvTotalCoins"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/total_coins_stats"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14"
+                        />
+                <TextView
+                        android:id="@+id/tvTotalCoinsValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="40233"
+                        android:textStyle="bold"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                        android:text="@string/total_coins_stats"
+                        android:textColor="#000000"
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14"
+                        />
+                <TextView
+                        android:id="@+id/tvTotalCoinsValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="40233"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: Không fix cứng, nếu để preview để xem khi code thì dùng tools:text

File: app/src/main/res/layout/fragment_home.xml [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+                    android:layout_width="@dimen/dp_0"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:orientation="horizontal">
+                <TextView
+                        android:id="@+id/tvTotalMarketCap"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/total_market_cap_stats"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14" />
+                <TextView
+                        android:id="@+id/currencySymbolTotalMarketCapValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="$"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+                        android:layout_height="wrap_content"
+                        android:text="@string/total_market_cap_stats"
+                        android:textColor="#000000"
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14" />
+                <TextView
+                        android:id="@+id/currencySymbolTotalMarketCapValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="$"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: không hard code, đặt vào resource

File: app/src/main/res/layout/fragment_home.xml [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textColor="#000000"
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14"
+                        android:layout_marginStart="@dimen/dp_5"/>
+                <TextView
+                        android:id="@+id/tvTotalMarketCapValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="2434.2B"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: Không HARD CODE

File: app/src/main/res/layout/fragment_home.xml [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+                    android:layout_width="@dimen/dp_0"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:orientation="horizontal">
+                <TextView
+                        android:id="@+id/tvTotalMarketplace"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/total_marketplace_stats"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14" />
+                <TextView
+                        android:id="@+id/tvTotalMarketplaceValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="40233"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+                    android:layout_width="@dimen/dp_0"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:orientation="horizontal">
+                <TextView
+                        android:id="@+id/tvVol24h"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/vol_24h_stats"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14" />
+                <TextView
+                        android:id="@+id/currencySymbolVol24hValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="$"
+                        android:textColor="#000000"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_home.xml [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+                        android:layout_marginStart="@dimen/dp_5"
+                        android:textStyle="bold"
+                        android:textSize="@dimen/sp_14" />
+                <TextView
+                        android:id="@+id/tvVol24hValue"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="63.8B"
+                        android:textColor="#000000"
+                        android:layout_marginStart="@dimen/dp_1"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]: .

File: app/src/main/res/layout/fragment_search.xml [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+            android:id="@+id/searchBar"
+            android:layout_width="@dimen/dp_0"
+            android:layout_height="wrap_content"
+            android:background="@drawable/bold_border_layout"
+            app:layout_constraintStart_toEndOf="@id/btnBack"
+            app:layout_constraintEnd_toEndOf="@id/topBarLayout"
+            app:layout_constraintTop_toTopOf="@id/topBarLayout"
+            app:layout_constraintBottom_toBottomOf="@+id/topBarLayout"
+            android:layout_margin="@dimen/dp_5">
+            <EditText
```
Comment (ghost) [inline app/src/main/res/layout/fragment_search.xml:43]: Các thẻ cách nhau 1 dòng trống
```suggestion
            
            <EditText
```

File: app/src/main/res/layout/fragment_search.xml [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+                android:id="@+id/search_edit_text"
+                android:layout_width="@dimen/dp_0"
+                android:layout_height="@dimen/dp_35"
+                android:hint="@string/search_hint"
+                android:background="@drawable/rounded_border_layout"
+                android:padding="@dimen/dp_0"
+                android:textColorHint="@android:color/darker_gray"
+                android:textSize="15sp"
+                android:inputType="text"
```
Comment (ghost) [inline app/src/main/res/layout/fragment_search.xml]: HARD CODE

File: app/src/main/res/layout/fragment_search.xml [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/bottomNavigationView"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:menu="@menu/bottom_navigation_menu"
+        app:labelVisibilityMode="unlabeled"/>
```
Comment (ghost) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
        app:labelVisibilityMode="unlabeled"/>
        
```

File: app/src/main/res/layout/fragment_search.xml [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+                app:layout_constraintTop_toTopOf="@id/searchBar"
+                app:layout_constraintBottom_toBottomOf="@+id/searchBar"
+                android:layout_marginTop="@dimen/dp_5"
+                android:layout_marginBottom="@dimen/dp_5"
+                android:layout_marginEnd="@dimen/dp_5"
+                />
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+
+
```
Comment (ghost) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
```

File: app/src/main/res/layout/item_coin.xml [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginEnd="@dimen/dp_15" />
+
+    <TextView
+        android:id="@+id/coinSymbol"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="BTC"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: HARD CODE

File: app/src/main/res/layout/item_coin.xml [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintStart_toEndOf="@id/coinIcon"
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginStart="@dimen/dp_8"
+        />
+
+    <TextView
+        android:id="@+id/coinName"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Bitcoin"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintStart_toEndOf="@id/coinIcon"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginStart="@dimen/dp_8"
+        />
+
+    <TextView
+        android:id="@+id/currencySymbolMarket"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="$"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+        android:textSize="@dimen/sp_14"
+        app:layout_constraintStart_toEndOf="@id/marketCap"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginStart="@dimen/dp_3"/>
+    <TextView
+        android:id="@+id/marketCapValue"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="1255.5B"
+        android:textSize="@dimen/sp_14"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 24]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintTop_toTopOf="parent"
+        />
+
+    <TextView
+        android:id="@+id/currencySymbolPrice"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="$"
+        android:textSize="@dimen/sp_16"
+        android:textStyle="bold"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 25]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintEnd_toStartOf="@id/coinPrice"
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginTop="@dimen/dp_1"
+        android:layout_marginEnd="@dimen/dp_1"/>
+    <TextView
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml:99]:
```suggestion
    
    <TextView
```

File: app/src/main/res/layout/item_coin.xml [hunk 26]
```diff
@@ -0,0 +1,167 @@
...
+        android:id="@+id/coinPrice"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="63604.67"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 27]
```diff
@@ -0,0 +1,167 @@
...
+        android:layout_marginTop="@dimen/dp_1"
+        android:layout_marginEnd="@dimen/sp_5"/>
+
+    <TextView
+        android:id="@+id/coinPriceChange"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="-0.70%"
+        android:textSize="@dimen/sp_14"
+        android:textColor="@android:color/holo_red_light"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 28]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintEnd_toStartOf="@id/currencySymbolPrice"
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginTop="@dimen/dp_3"
+        android:layout_marginEnd="@dimen/dp_10"/>
+    <ImageView
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml:121]:
```suggestion
  
    <ImageView
```

File: app/src/main/res/layout/item_coin.xml [hunk 29]
```diff
@@ -0,0 +1,167 @@
...
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginTop="@dimen/dp_6"
+        android:layout_marginEnd="@dimen/sp_5"/>
+
+
+    <TextView
+            android:id="@+id/currencySymbolVolumeValue"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="$"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 30]
```diff
@@ -0,0 +1,167 @@
...
+            android:textSize="@dimen/sp_14"
+            app:layout_constraintEnd_toStartOf="@id/volumeValue"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:layout_marginEnd="@dimen/dp_1"/>
+    <TextView
+        android:id="@+id/volumeValue"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="28.1B"
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]: .

File: app/src/main/res/layout/item_coin.xml [hunk 31]
```diff
@@ -0,0 +1,167 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/volume"
+        android:textSize="@dimen/sp_14"
+        android:textColor="@android:color/darker_gray"
+        app:layout_constraintEnd_toStartOf="@id/currencySymbolVolumeValue"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginEnd="@dimen/dp_3"/>
+
+
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]:
```suggestion

```

File: app/src/main/res/menu/bottom_navigation_menu.xml [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+
+    <item
+        android:id="@+id/settings"
+        android:title="@string/setting"
+        android:icon="@drawable/ic_setting"
+        />
+
+
+
+
```
Comment (ghost) [inline app/src/main/res/menu/bottom_navigation_menu.xml]:
```suggestion

```

File: app/src/main/res/menu/bottom_navigation_menu.xml [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+</menu>
```
Comment (ghost) [inline app/src/main/res/menu/bottom_navigation_menu.xml]: End of file
```suggestion
</menu>

```

File: app/src/main/res/values-vi/strings.xml [hunk 34]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources></resources>
```
Comment (ghost) [inline app/src/main/res/values-vi/strings.xml]: Sao lại trống trơn thế này

**Review by ghost — CHANGES_REQUESTED** [review]
CODE QUÁ ẨU

File: app/src/main/res/layout/fragment_home.xml [hunk 35]
```diff
@@ -0,0 +1,223 @@
...
+                    android:textSize="@dimen/sp_14" />
+
+                <TextView
+                    android:id="@+id/tvTotalCoinsValue"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textStyle="bold"
+                    android:textColor="@color/black"
+                    android:layout_marginStart="@dimen/dp_5"
+                    android:textSize="@dimen/sp_14" />
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml:63]: Các thẻ cách nhau 1 dòng trống
```suggestion
                    android:textSize="@dimen/sp_14" />
                    
```

File: app/src/main/res/layout/fragment_home.xml [hunk 36]
```diff
@@ -0,0 +1,223 @@
...
+                <TextView
+                    android:id="@+id/tvTotalMarketCapValue"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textColor="@color/black"
+                    android:textStyle="bold"
+                    android:textSize="@dimen/sp_14"
+                    android:layout_marginStart="@dimen/dp_1" />
+            </LinearLayout>
+        </LinearLayout>
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]:
```suggestion
                    android:layout_marginStart="@dimen/dp_1" />
                    
            </LinearLayout>
            
        </LinearLayout>
```

File: app/src/main/res/layout/fragment_home.xml [hunk 37]
```diff
@@ -0,0 +1,223 @@
...
+                    android:textSize="@dimen/sp_14" />
+
+                <TextView
+                    android:id="@+id/tvTotalMarketplaceValue"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textColor="@color/black"
+                    android:textStyle="bold"
+                    android:textSize="@dimen/sp_14"
+                    android:layout_marginStart="@dimen/dp_5" />
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml:132]:
```suggestion
                    android:layout_marginStart="@dimen/dp_5" />
                    
```

File: app/src/main/res/layout/fragment_home.xml [hunk 38]
```diff
@@ -0,0 +1,223 @@
...
+                    android:id="@+id/tvVol24hValue"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textColor="@color/black"
+                    android:layout_marginStart="@dimen/dp_1"
+                    android:textStyle="bold"
+                    android:textSize="@dimen/sp_14" />
+            </LinearLayout>
+        </LinearLayout>
+    </LinearLayout>
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml]:
```suggestion
                    android:textSize="@dimen/sp_14" />
                    
            </LinearLayout>
            
        </LinearLayout>
        
    </LinearLayout>
```

File: app/src/main/res/layout/fragment_home.xml [hunk 39]
```diff
@@ -0,0 +1,223 @@
...
+            app:layout_constraintTop_toTopOf="@+id/filterLayout"
+            app:layout_constraintBottom_toBottomOf="@id/filterLayout" />
+
+        <Spinner
+            android:id="@+id/spinnerOrderBy"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintEnd_toEndOf="@+id/filterLayout"
+            app:layout_constraintTop_toTopOf="@+id/filterLayout"
+            app:layout_constraintBottom_toBottomOf="@id/filterLayout" />
```
Comment (ghost) [inline app/src/main/res/layout/fragment_home.xml:208]:
```suggestion
            app:layout_constraintBottom_toBottomOf="@id/filterLayout" />
            
```

File: app/src/main/res/layout/fragment_search.xml [hunk 40]
```diff
@@ -0,0 +1,96 @@
...
+                android:src="@drawable/ic_search"
+                android:scaleType="fitXY"
+                android:background="?attr/selectableItemBackgroundBorderless"
+                app:layout_constraintEnd_toEndOf="@id/searchBar"
+                app:layout_constraintTop_toTopOf="@id/searchBar"
+                app:layout_constraintBottom_toBottomOf="@id/searchBar"
+                android:layout_marginTop="@dimen/dp_5"
+                android:layout_marginBottom="@dimen/dp_5"
+                android:layout_marginEnd="@dimen/dp_5" />
+        </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (ghost) [inline app/src/main/res/layout/fragment_search.xml]:
```suggestion
                android:layout_marginBottom="@dimen/dp_5"
                android:layout_marginEnd="@dimen/dp_5" />
                
        </androidx.constraintlayout.widget.ConstraintLayout>
```

File: app/src/main/res/layout/item_coin.xml [hunk 41]
```diff
@@ -0,0 +1,157 @@
...
+    <TextView
+        android:id="@+id/currencySymbolMarket"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="@dimen/sp_14"
+        app:layout_constraintStart_toEndOf="@id/marketCap"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginStart="@dimen/dp_3"/>
+    <TextView
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]:
```suggestion
        app:layout_constraintBottom_toBottomOf="parent"
        android:layout_marginStart="@dimen/dp_3"/>
        
    <TextView
```

File: app/src/main/res/layout/item_coin.xml [hunk 42]
```diff
@@ -0,0 +1,157 @@
...
+
+    <TextView
+            android:id="@+id/currencySymbolVolumeValue"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="@dimen/sp_14"
+            app:layout_constraintEnd_toStartOf="@id/volumeValue"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:layout_marginEnd="@dimen/dp_1"/>
+    <TextView
```
Comment (ghost) [inline app/src/main/res/layout/item_coin.xml]:
```suggestion
            android:layout_marginEnd="@dimen/dp_1"/>
            
    <TextView
```

**Review by ghost — CHANGES_REQUESTED** [review]
check my cmt

**Review by ghost — APPROVED** [review]

**Pull request merged**